io-engine/src/subsys/config/opts.rs

```diff
@@ -94,11 +181,13 @@ pub struct NvmfTgtConfig {
     /// NVMF target Command Retry Delay in x100 ms.
     pub crdt: [u16; TARGET_CRDT_LEN],
     /// TCP transport options
-    pub opts: NvmfTcpTransportOpts,
+    pub opts_tcp: NvmfTransportOpts,
     /// NVMF target interface (ip, mac, name or subnet).
     pub interface: Option<String>,
     /// Enable RDMA for NVMF target or not
     pub rdma: Option<bool>,
+    /// RDMA transport options
+    pub opts_rdma: NvmfTransportOpts,
```

Comment:
I suggest having a different way, example:
rdmaOpts:
  field: env_var("NVMF_RDMA_....", env_var("NVMF", rdma default ? common default
So, first check for transport specific override, then generic override, then specific default, and only if nothing sets it then common default

Reply:
what if the env has both transport specific overrides?
On the other hand, most of these opts are ideally not supposed to be transport specific(except those which are) but since there was only TCP till now, we have the envs named specific to TCP.

Reply:
> On the other hand, most of these opts are ideally not supposed to be transport specific(except those which are)

Which ones are transport specific?

> but since there was only TCP till now, we have the envs named specific to TCP.

Yes which is why I'm saying if some of those are also applicable for RDMA then do we need to have different env var names if we want to control them independently?

Reply:
> Which ones are transport specific?

e.g ack_timeout is valid only for TCP, and data_wr_size is valid only for RDMA.

> Yes which is why I'm saying if some of those are also applicable for RDMA then do we need to have different env var names if we want to control them independently?

I'd rather argue for renaming all current env variables generically without transport name, to lessen the circus required for setting up opts. If required to override separately for RDMA, do it via RDMA specific env from inside `for_rdma()`. Let's get on a call tomorrow to sort this out.

Reply:
addressed as discussed.